cmd/jira-lifecycle-plugin/server.go

```diff
@@ -539,13 +542,43 @@ Comment <code>/jira refresh</code> to re-evaluate validity if changes to the Jir
 		}
 	}
 
-	if needsValidLabel && !hasValidLabel {
-		if err := ghc.AddLabel(e.org, e.repo, e.number, labels.ValidBug); err != nil {
-			log.WithError(err).Error("Failed to add valid bug label.")
+	if hasValidBZLabel && !needsValidLabel {
+		humanLabelled, err := ghc.WasLabelAddedByHuman(e.org, e.repo, e.number, labels.BugzillaValidBug)
+		if err != nil {
+			// Return rather than potentially doing the wrong thing. The user can re-trigger us.
+			return fmt.Errorf("failed to check if %s label was added by a human: %w", labels.BugzillaValidBug, err)
+		}
+		if humanLabelled {
+			// This will make us remove the invalid label if it exists but saves us another check if it was
+			// added by a human. It is reasonable to assume that it should be absent if the valid label was
+			// manually added.
+			needsInvalidLabel = false
+			needsValidLabel = true
+			response += fmt.Sprintf("\n\nRetaining the %s label as it was manually added.", labels.BugzillaValidBug)
 		}
-	} else if !needsValidLabel && hasValidLabel {
-		if err := ghc.RemoveLabel(e.org, e.repo, e.number, labels.ValidBug); err != nil {
-			log.WithError(err).Error("Failed to remove valid bug label.")
+	}
+
+	if needsValidLabel {
+		if !hasValidJiraLabel {
+			if err := ghc.AddLabel(e.org, e.repo, e.number, labels.ValidBug); err != nil {
+				log.WithError(err).Error("Failed to add valid bug label.")
+			}
+		}
+		if !hasValidBZLabel {
+			if err := ghc.AddLabel(e.org, e.repo, e.number, labels.BugzillaValidBug); err != nil {
+				log.WithError(err).Error("Failed to add valid bugzilla bug label.")
+			}
+		}
+	} else if !needsValidLabel {
```

Comment:
Is the `if !needsValidLabel` really necessary here?

Reply:
Nope. Changed to a simple `else` statement